docs/big-graph-from-cypher.adoc

```diff
@@ -0,0 +1,248 @@
+
+[source,bash,subs=attributes]
+----
+./performanceCypherTest.sh 'neo4jHome' 'userName' 'password' 'address'
+----
+
+the `address` parameter  is optional, the default address is : bolt://localhost:7687
+If you use the LDBC SF1 graph, or another big one is better to change the `open files allowed` from the default 1024 at last to 40.000.
+
+Download link:{script}/performanceCypherTest.sh[performanceCypherTest.sh]
+
+== Export all
+
+=== Batch size
+
+With the use of the config param `batchSize` we done some tests with different batch size.
+The default value is 20000.
+
+[source,bash,subs=attributes]
+----
+time ./cypher-shell -u yourUsername -p yourPassword
+"call apoc.export.cypher.all('yourPath/exportAll', {format:'neo4j-shell', batchSize: 10000})"
+----
+
+==== Results
+
+.Default 20.000
+
+----
+real 1m52.744s
+user 0m0.936s
+sys 0m0.064s
+----
+
+.Batch size 10.000
+
+----
+real    1m50.715s
+user    0m0.932s
+sys     0m0.076s
+----
+
+.Batch size 1.000
+
+----
+real    1m49.577s
+user    0m0.888s
+sys     0m0.120s
+----
+
+.Batch size 100
+
+----
+real    1m51.297s
+user    0m0.928s
+sys     0m0.088s
+----
+
+=== Different output formats
+
+We try the different output formats, changing the `config` parameter `format`.
+
+[source,bash,subs=attributes]
+----
+time ./cypher-shell -u yourUsername -p yourPassword  
+"call apoc.export.cypher.all('yourPath/exportData.cypher', {format:'neo4j-shell'})"
+----
+
+==== Results
+
+.neo4j-shell
+
+----
+real    1m49.268s
+user    0m0.904s
+sys     0m0.072s
+----
+
+.cypher-shell
+
+----
+real    1m55.089s
+user    0m0.892s
+sys     0m0.092s
+----
+
+.plain
+
+----
+real    1m54.490s
+user    0m0.932s
+sys     0m0.076s
+----
+
+=== Many files (separateFiles config)
+
+With the param `separateFiles` (default false) we can export our graph or part of it, in different files.
+In the example below we name the exported file `exportAll.cypher` so our export will be:
+
+ * exportAll.cleanup.cypher
+ * exportAll.nodes.cypher
+ * exportAll.relationships.cypher
+ * exportAll.schema.cypher
+
+[source,bash,subs=attributes]
+----
+time ./cypher-shell -u yourUsername -p yourPassword  
+"call apoc.export.cypher.all('yourPath/exportAll.cypher', {format:'neo4j-shell',separateFiles:true})"
+----
+
+==== Result
+
+----
+real 1m55.229s
+user 0m0.960s
+sys  0m0.084s
+----
+
+
+== Export from query
+
+[source,bash,subs=attributes]
+----
+time ./cypher-shell -u yourUsername -p yourPassword
+"call apoc.export.cypher.query('MATCH (n) OPTIONAL MATCH p = (n)-[r]-(m) RETURN n,r,m',
+'yourPath/exportQuery.cypher', {format:'neo4j-shell', batchSize: 10000})"
+----
+
+=== Result
+
+----
+real    3m34.924s
+user    0m0.992s
+sys     0m0.068s
+----
+
+== Export from given nodes and rels
+
+[source,bash,subs=attributes]
+----
+time ./cypher-shell -u yourUsername -p yourPassword  "Match (n:Person)-[r:LIKES_COMMENT]->(c:Comment) CALL apoc.expor
```

Comment:
this is wrong, it needs to collect the nodes and rels into 2 collections.
can you check the output file of this?